src/modules/ekf2/EKF/estimator_interface.cpp

```diff
@@ -490,14 +490,16 @@ bool EstimatorInterface::initialise_interface(uint64_t timestamp)
 		max_time_delay_ms = math::max(_params.ev_delay_ms, max_time_delay_ms);
 	}
 
-	// calculate the IMU buffer length required to accommodate the maximum delay with some allowance for jitter
-	_imu_buffer_length = ceilf(max_time_delay_ms / FILTER_UPDATE_PERIOD_MS);
+	const float filter_update_period_ms = _params.filter_update_interval_us / 1000.f;
+
+	// calculate the IMU buffer length required to accomodate the maximum delay with some allowance for jitter
+	_imu_buffer_length = ceilf(max_time_delay_ms / (filter_update_period_ms)) + 1;
```

Comment:
Adding 1 does allow some margin but will also result in a slightly more delayed EKF fusion horizon that does slightly degrade output observer tracking accuracy.

Reply:
removed